paddle/fluid/platform/dynload/warpctc.h

```diff
@@ -31,10 +31,15 @@ extern void* warpctc_dso_handle;
  * (for each function) to dynamic load warpctc routine
  * via operator overloading.
  */
+#ifndef _WIN32
+#define DECLARE_TYPE(__name, ...) ->decltype(__name(__VA_ARGS__))
+#else
+#define DECLARE_TYPE(__name, ...)
+#endif
 #define DYNAMIC_LOAD_WARPCTC_WRAP(__name)                                      \
   struct DynLoad__##__name {                                                   \
     template <typename... Args>                                                \
-    auto operator()(Args... args) -> decltype(__name(args...)) {               \
+    auto operator()(Args... args) DECLARE_TYPE(__name, args...) {              \
```

Comment:
By the way, did you test the warpctc when dso is turn off?  I'm double about if it can works.

Reply:
DSO is ON by default for warp-ctc, seems they forget to set it when introduct the DSO into warpctc.